### ISSUE TITLE
enforceUnique param for Reactions is getting ignored

### ISSUE THREAD

**Issue by tomSidebar** [GetStream/stream-chat-swift#1856]
## What did you do?
We want to enforce unique message reactions in our app, following https://getstream.io/chat/docs/ios-swift/send_reaction/?language=swift.

We created our own ChatMessageReactionsVC class:
```
class SBChatMessageReactionsVC: ChatMessageReactionsVC {

    // toggleReaction toggles on/off the reaction for the message
    override func toggleReaction(_ reaction: MessageReactionType) {
        guard let message = messageController.message else { return }

        let completion: (Error?) -> Void = { [weak self] _ in
            self?.dismiss(animated: true)
        }

        let shouldRemove = message.currentUserReactions.contains { $0.type == reaction }
        shouldRemove
            ? messageController.deleteReaction(reaction, completion: completion)
            : messageController.addReaction(reaction, enforceUnique: true, completion: completion)
    }
}
```

We're using `Components.default.reactionPickerVC = SBChatMessageReactionsVC.self` to instantiate our custom class. We verified via debugging that the custom class is getting called when a new reaction is added and that the enforceUnique param is correctly set to true.

## What did you expect to happen?
Have reactions limited to one per user per message.

## What happened instead?
Can still use multiple reactions on a single message.

## GetStream Environment
**GetStream Chat version: 4.10.0**
**GetStream Chat frameworks:** StreamChat, StreamChatUI
**iOS version:**
**Swift version:**
**Xcode version:**
**Device:**

**Comment by polqf** [comment]
Hi @tomSidebar,
I managed to reproduce this issue, so I went ahead and filled an internal ticket.

We'll try to fix this asap, and will come back to you once fixed.

**Comment by polqf** [comment]
After reviewing this with our backend team, we found out that our documentation has a mistake.
`enforce_unique` does not refer to only have one reaction per message per user, but instead to have one reaction per type.
We are working on updating the documentation now.

With our SDK you can "like" a message multiple times if you want, and `enforce_unique` would make sure that only one "like" is set.
Sorry for the misunderstanding

To accomplish the result you want, you will need to remove the existing reaction, and add the new one.

**Comment by polqf** [comment]
Working on a fix here: https://github.com/GetStream/stream-chat-swift/pull/1861/

**Comment by polqf** [comment]
Hi Tom. There's no reason why to think React did the same mistake 😄

**Comment by petyosi** [comment]
@tomSidebar we are tracking that implementation for the React SDK as well, see linked issue.

**Comment by nuno-vieira** [comment]
Hi @tomSidebar!

This is now available on the [4.13.0](https://github.com/GetStream/stream-chat-swift/releases/tag/4.13.0) release.

Thank you!
